tests/post/steps/test_authentication.py

```diff
@@ -0,0 +1,207 @@
+import json
+import re
+
+import requests
+import requests.exceptions
+
+import pytest
+from pytest_bdd import scenario, given, then, when, parsers
+
+import kubernetes.client
+from kubernetes.client.rest import ApiException
+
+from tests import kube_utils
+from tests import utils
+
+
+# Constants {{{
+
+INGRESS_PORT = 8443
+
+# }}}
+
+
+# Scenarios {{{
+
+@scenario('../features/authentication.feature', 'List Pods')
+def test_list_pods(host):
+    pass
+
+
+@scenario('../features/authentication.feature', 'Expected Pods')
+def test_expected_pods(host):
+    pass
+
+
+@scenario('../features/authentication.feature', 'Reach the OpenID Config')
+def test_reach_openid_config(host):
+    pass
+
+
+@scenario('../features/authentication.feature', 'Access HTTPS service')
+def test_access_https_service(host):
+    pass
+
+
+@scenario('../features/authentication.feature',
+          'Login to Dex using incorrect email')
+def test_failed_login(host):
+    pass
+
+
+@scenario('../features/authentication.feature',
+          'Login to Dex using correct email and password')
+def test_login(host):
+    pass
+
+
+@pytest.fixture(scope='function')
+def context():
+    return {}
+
+# }}}
+
+# Given {{{
+
+# }}}
+
+
+# When {{{
+
+
+@when(parsers.parse(
+    "we perform a request on '{path}' with port '{port}' on control-plane IP"))
+def perform_request(host, context, path, port):
+    ip = _get_control_plane_ip(host)
+    try:
+        context['response'] = requests.get(
+            'https://{ip}:{port}{path}'.format(
+                ip=ip, port=port, path=path
+            ),
+            verify=False,
+        )
+    except requests.exceptions.ConnectionError as exc:
+        context['exception'] = exc
+
+
+@when(parsers.parse(
+    "we login to Dex as '{username}' using password '{password}'"))
+def dex_login(host, username, password, context):
+    ip = _get_control_plane_ip(host)
+    context['login_response'] = _dex_auth_request(
+        ip, username, password
+    )
+
+
+# }}}
+
+
+# Then {{{
+
+
+@then("we can reach the OIDC openID configuration")
+def reach_openid_config(host):
+    ip = _get_control_plane_ip(host)
+    try:
+        response = requests.get(
+            'https://{ip}:8443/oidc/.well-known/openid-configuration'.format(
+                ip=ip
+            ),
+            verify=False,
+        )
+    except requests.exceptions.ConnectionError as exc:
+        raise ConnectionError(exc)
```

Comment:
Is `ConnectionError` in stdlib? What does it mean? I'd rather have a `pytest.fail` here.